internal/bsr/container.go

```diff
@@ -514,3 +538,42 @@ func (c *container) close(_ context.Context) error {
 
 	return closeError
 }
+
+// ValidateChecksums iterates over all files in the SHA256SUM
+// file and verifies that each file checksum is as expected.
+//
+// This function expects that the container's kms keys
+// are loaded into memory and the signature files are
+// verified.
+func (c *container) ValidateChecksums(ctx context.Context) (ContainerChecksumValidation, error) {
+	const op = "bsr.(container).Validate"
+	if len(c.shaSums) == 0 {
+		return nil, fmt.Errorf("%s: missing checksums", op)
+	}
+	if c.keys == nil {
+		return nil, fmt.Errorf("%s: missing keys", op)
+	}
+	checksumValidation := make(ContainerChecksumValidation, len(c.shaSums))
+	for fileName, expectedChecksum := range c.shaSums {
+		report := &FileChecksumValidation{
+			Filename: fileName,
+		}
+		checksumValidation[fileName] = report
+		f, err := c.container.OpenFile(ctx, fileName)
+		if err != nil {
+			report.Error = err
+			continue
+		}
+		actualChecksum, err := crypto.Sha256Sum(ctx, f, crypto.WithHexEncoding(true))
+		if err != nil {
+			report.Error = err
+			continue
+		}
```

Comment:
We should call `f.Close()` after reading the contents of the file. It might be helpful to move this logic of opening, reading, and calculating the checksum for a single file into a function, so that you can use a `defer` to ensure the file is closed before moving on to the next file.